### ISSUE TITLE
upgrade to 27.1?

### ISSUE THREAD

**Issue by otterpro** [enko/EmacsChocolateyPackage#16]
Can we upgrade to the latest version v27.1?  Thanks a bunch.

**Comment by enko** [comment]
No binaries yet:

https://ftp.gnu.org/pub/gnu/emacs/windows/

**Comment by michaelmhoffman** [comment]
Pretest versions of Emacs 27.1 Windows binaries are now available: https://alpha.gnu.org/gnu/emacs/pretest/windows/emacs-27/

**Comment by enko** [comment]
@michaelmhoffman Thanks a lot! Will try to build this evening (in ~10 hours from now).

**Comment by enko** [comment]
Work is going on in #17

**Comment by enko** [comment]
I have uploaded the pretest here: https://chocolatey.org/packages/Emacs/27.1.0.20200820

**Comment by michaelmhoffman** [comment]
I tried upgrading and got the following error:

```
> choco upgrade emacs
Chocolatey v0.10.15
Upgrading the following packages:
emacs
By upgrading you accept licenses for the packages.

You have emacs v26.3.0.20191219 installed. Version 27.1.0.20200820 is available based on your source(s).
Progress: Downloading Emacs 27.1.0.20200820... 100%

Emacs v27.1.0.20200820 [Approved]
emacs package files upgrade completed. Performing other installation steps.
Downloading emacs 64 bit
  from 'https://alpha.gnu.org/gnu/emacs/pretest/windows/emacs-27/emacs-27.1-x86_64-no-deps.zip'
Progress: 100% - Completed download of C:\Users\mhoffman\AppData\Local\Temp\chocolatey\Emacs\27.1.0.20200820\emacs-27.1-x86_64-no-deps.zip (47.15 MB).
Download of emacs-27.1-x86_64-no-deps.zip (47.15 MB) completed.
Hashes match.
Extracting C:\Users\mhoffman\AppData\Local\Temp\chocolatey\Emacs\27.1.0.20200820\emacs-27.1-x86_64-no-deps.zip to C:\ProgramData\chocolatey\lib\Emacs\tools/emacs...
ERROR: Can not delete output file : The filename, directory name, or volume label syntax is incorrect. : C:\ProgramData\chocolatey\lib\Emacs\tools/emacs\bin\libXpm-noX4.dll
ERROR: 7-Zip encountered a fatal error while extracting the files. This is most likely an issue with the 'Emacs' package and not with Chocolatey itself. Please follow up with the package maintainer(s) directly.
The upgrade of emacs was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\Emacs\tools\chocolateyInstall.ps1'.
 See log for details.

Chocolatey upgraded 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - emacs (exited 2) - Error while running 'C:\ProgramData\chocolatey\lib\Emacs\tools\chocolateyInstall.ps1'.
 See log for details.
```

The section of chocolatey.log is too long to add to this comment. I can put it on a gist if you need it.

**Comment by enko** [comment]
While trying to debug this I got hit by #18 (Bad Checksum) and also the deps package got missing. I depublished the 27.1 chocolatey for now and will look into it tommorow.

Pretest seems to be moving target still.

**Comment by enko** [comment]
So a final release has been made and it is just the installer:

https://ftp.gnu.org/pub/gnu/emacs/windows/emacs-27/

Will try to integrate that teh coming days.

**Comment by notuntoward** [comment]
To be a pest, will this be on choco soon?  Thanks.

**Comment by angrybacon** [comment]
Is the newly acquired JSON native support a thing on Windows? On macOS, it only takes a `brew install jansson` (or equivalent) with the port I'm using (`emacs-plus`).

**Comment by enko** [comment]
Hey folks,

I have uploaded a new release, could you please try the new release.

https://chocolatey.org/packages/Emacs/27.1.0.20200909

**Comment by notuntoward** [comment]
Thanks.  I downloaded it, deleted my .emacs.d and let it rebuild all the packages in my .emacs file.  But there still is some kind of public key problem.  Here's what I see in my emacs \*Error\* buffer:

> Failed to verify signature archive-contents.sig:
> No public key for 066DAFCB81E42C40 created at 2020-09-10T14:05:02-0700 using RSA
> Command output:
> gpg: keyblock resource '/c/Users/scott/c:/Users/scott/.emacs.d/elpa/gnupg/pubring.kbx': No such file or directory
> gpg: Signature made Thu Sep 10 14:05:02 2020 PDT
> gpg:                using RSA key C433554766D3DDC64221BFAA066DAFCB81E42C40
> gpg: Can't check signature: No public key

**Comment by enko** [comment]
@notuntoward I started `emacs -Q` and then installed the `docbook` package via `package-list-packages` and it worked? 

@angrybacon at least `libjansson-4.dll` is included in the distribution and it seems to work:

```
(json-parse-string "{\"Hello\":\"World\"}")
#s(hash-table size 1 test equal rehash-size 1.5 rehash-threshold 0.8125 data ("Hello" "World"))
```

**Comment by tekai** [comment]
Emacs 27.2 has been released and more critically the location for 27.1 changed, it's currently not possible to install emacs.

**Comment by enko** [comment]
@tekai I tried that yesterday and the process worked fine? Looking at https://ftp.gnu.org/pub/gnu/emacs/windows/emacs-27/ the URL still follow the same pattern?
